test/tests/functional/pbs_pbsnodes_output_trimmed.py

```diff
@@ -0,0 +1,84 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+import os
+import subprocess
+from subprocess import Popen,PIPE
+
+class TestPbsnodesOutputTrimmed(TestFunctional):
+        """
+        This test suite tests pbsnodes executable with -l
+	and makes sure that the node names are not trimmed.
+        """
+	def test_pbsnodes_output(self):
+	        """
+		This method adds a new entry to /etc/hosts and creates
+		a mom with name more than 20 characters, then makes the
+		node offline and checks the pbsnodes -l output and
+		makes sure it is not trimmed
+		"""
+		pbsnodes = os.path.join(str(self.server.pbs_conf['PBS_EXEC']), "bin", "pbsnodes")
+		hname = ['123456789012345678901234567890', '12345678901234567890', '123456789012345']
+		tmpip = ['127.0.0.1', '127.0.0.2', '127.0.0.3']
+		hosts = open("/etc/hosts","a")
+		for index in range(3):
+		        tmpentry = tmpip[index] + " " + hname[index] + " " + hname[index]
+		        hosts.write(str(tmpentry));
```

Comment:
You plussed tmpentry together in the line above.  If it wasn't a string, python would complain when you tried to plus non-strings together.  You don't need to str() this.